vncviewer/vncviewer.cxx

```diff
@@ -368,6 +369,56 @@ static void usage(const char *programName)
   exit(1);
 }
 
+static void
+potentiallyLoadConfigurationFile(char *vncServerName)
+{
+  struct stat st;
+  if (stat(vncServerName, &st) != 0) {
+    const int errno_ = errno;
+    switch (errno_) {
+    case EACCES:
+    case ELOOP:
+    case ENOMEM:
+    case EOVERFLOW:
+      vlog.error("%s: %s", vncServerName, strerror(errno_));
+      if (alertOnFatalError)
+        fl_alert("%s: %s", vncServerName, strerror(errno_));
+      exit(EXIT_FAILURE);
+    default:
+      // Such as ENAMETOOLONG, ENOTDIR, ENOENT, ENXIO, etc.
+      vlog.debug("%s: errno(%d) %s, it might be a server",
+                 vncServerName, errno_, strerror(errno_));
+      break;
+    }
+  }
+  else {
+    if (S_ISREG(st.st_mode)) {
+#ifndef WIN32
+      const bool absolute = vncServerName[0] == '/';
+#else
+      const bool absolute = vncServerName[0] == '\\' || vncServerName[1] == ':';
+#endif
+      if (!absolute && vncServerName[0] != '.') {
```

Comment:
You can have normal files that start with `.` (and maybe even a hostname?).
I suggest you simply look for a path separator (`\` on Windows and `/` on Unix). No hostname can contain those. So you can do that check first and avoid the `stat()` call completely.

Reply:
Thank you for the comment! I like the comment about `/` and `\`, but `stat()` also serves the purpose to make sure the file exists and that the user has the access, like you proposed in previous attempt. Need to establish that knowledge before calling `loadViewerParameters()`. I can return back to `fopen()` and make some special handling for the permission denied case, if you like?

Reply:
What extra error handling do we actually need? Just call `loadViewerParameters()` and handle any exception. It will tell you why the file couldn't be opened.

Reply:
Originally I wanted to dodge any failures in `fopen()` in the event that the given argument in fact was a host. However, with your latest comment about checking for the path separator, we can distinguish if we have a file or hostname directly (or at least supposed to be). I understand now and will change accordingly, thank you for the persistently good code review comments! Please bear with me :)